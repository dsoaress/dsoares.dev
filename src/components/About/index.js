import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import BlockContent from '../BlockContent'
import Social from '../Social'
import * as S from './styled'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <S.Wrapper>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <S.Grid>
        <S.TextWrapper>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </S.TextWrapper>
        <S.Image fluid={sanitySiteSettings.bio.image.asset.fluid} />
      </S.Grid>
    </S.Wrapper>
  )
}

export default About
