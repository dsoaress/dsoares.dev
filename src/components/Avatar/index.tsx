import Image from 'next/image'

import avatar from '@/assets/avatar.jpg'
import profileData from '@/content/profile.json'

import { Overlay, Wrapper } from './styles'

export function Avatar() {
  return (
    <Wrapper>
      <Image
        src={avatar}
        alt={profileData.title}
        objectFit="cover"
        objectPosition="top"
        layout="fill"
        placeholder="blur"
        quality={100}
      />
      <Overlay />
    </Wrapper>
  )
}
