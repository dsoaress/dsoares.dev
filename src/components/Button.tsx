import cn from 'classnames'
import type { ButtonHTMLAttributes } from 'react'

export function Button({ className, ...props }: ButtonHTMLAttributes<HTMLButtonElement>) {
  return (
    <button
      className={cn(
        'flex items-center px-6 text-sm font-medium leading-none transition-colors duration-300 border-0 rounded-md cursor-pointer h-11 bg-primary-600 text-neutral-50 hover:bg-primary-700 active:bg-primary-800',
        className
      )}
      {...props}
    />
  )
}
