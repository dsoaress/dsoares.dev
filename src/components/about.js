import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

import BlockContent from './block-content'
import Social from './social'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <div>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <div>
        <div>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </div>
        <Img fluid={sanitySiteSettings.bio.image?.asset.fluid} />
      </div>
    </div>
  )
}

export default About
