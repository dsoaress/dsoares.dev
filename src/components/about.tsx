import { Block } from './block'

export function About() {
  return (
    <Block className="col-span-12 text-3xl leading-snug">
      <p className="text-neutral-400">
        Specializing in <span className="text-neutral-50">scalable, robust applications</span>, I
        focus on delivering seamless user experiences and optimizing backend systems for maximum
        efficiency.
      </p>
    </Block>
  )
}
