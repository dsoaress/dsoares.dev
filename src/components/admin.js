import { useState } from 'react'
import useSWR, { mutate } from 'swr'
import { signOut } from 'next-auth/client'

import Spinner from '@/components/spinner'
import { fetcher } from '@/lib/fetcher'

export default function Tasks() {
  const [label, setLabel] = useState('')
  const [url, setUrl] = useState('')

  const { data } = useSWR('/api', fetcher, {
    refreshInterval: 500
  })

  if (!data) return <Spinner />

  async function handleSubmit(e) {
    e.preventDefault()
    mutate('/api', { ...data, data: { links: { label, url } } }, false)

    await fetch('/api', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ link: { label, url } })
    })

    mutate('/api')
    setLabel('')
    setUrl('')
  }

  return (
    <div>
      <div>
        <button onClick={() => signOut()}>Sign out</button>
      </div>
      <form onSubmit={handleSubmit}>
        <div className="text-black">
          <label>Label</label>
          <input
            type="text"
            value={label}
            onChange={e => setLabel(e.target.value)}
            required
          />
          <label>Url</label>
          <input
            type="text"
            value={url}
            onChange={e => setUrl(e.target.value)}
            required
          />
        </div>
        <button type="submit">Add</button>
      </form>
      <div>
        {!data.links || data.links.length === 0 ? (
          <p>
            It looks like you don't have any more tasks... Would you like to add
            a new one?
          </p>
        ) : (
          data.links.map(link => (
            <div key={link._id}>
              <p>{link.label}</p>
              <p>{link.url}</p>
            </div>
          ))
        )}
      </div>
    </div>
  )
}
