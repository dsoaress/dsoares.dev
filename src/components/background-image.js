import React from 'react'

const BackGroundImage = ({ children, className, image }) => (
  <div
    style={{
      backgroundImage: `url(${
        !!image.childImageSharp ? image.childImageSharp.fluid.src : image
      })`,
      backgroundPosition: `center center`,
      backgroundSize: `cover`
    }}
    className={className}
  >
    {children}
  </div>
)

export default BackGroundImage
