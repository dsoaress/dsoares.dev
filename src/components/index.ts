export * from './about'
export * from './block'
export * from './footer'
export * from './header'
export * from './home-container'
export * from './logo'
export * from './social'
