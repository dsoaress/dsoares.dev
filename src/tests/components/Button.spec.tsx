import { fireEvent, render, screen } from '@testing-library/react'

import { Button } from '@/components/Button'

describe('<Button />', () => {
  it('should matches snapshot', () => {
    render(<Button>Children</Button>)
    const element = screen.getByRole('button')

    expect(element).toMatchSnapshot()
  })

  it('should render with children', () => {
    render(<Button>Children</Button>)
    const element = screen.getByRole('button')

    expect(element).toHaveTextContent('Children')
  })

  it('should render with with additional className', () => {
    render(<Button className="test">Children</Button>)
    const element = screen.getByRole('button')

    expect(element).toHaveClass('test')
  })

  it('should call onClick handler', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick}>Children</Button>)
    const element = screen.getByRole('button')
    fireEvent.click(element)

    expect(onClick).toHaveBeenCalledTimes(1)
  })

  it('should render with disabled attribute', () => {
    render(<Button disabled>Children</Button>)
    const element = screen.getByRole('button')

    expect(element).toHaveAttribute('disabled')
  })

  it('should render with type attribute', () => {
    render(<Button type="submit">Children</Button>)
    const element = screen.getByRole('button')

    expect(element).toHaveAttribute('type', 'submit')
  })
})
