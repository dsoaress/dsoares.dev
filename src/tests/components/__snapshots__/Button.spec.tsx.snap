// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should matches snapshot 1`] = `
<button
  class="flex items-center px-6 text-sm font-medium leading-none transition-colors duration-300 border-0 rounded-md cursor-pointer h-11 bg-primary-600 text-neutral-50 hover:bg-primary-700 active:bg-primary-800"
>
  Children
</button>
`;
